src/Analyser/NamedArgumentsHelper.php

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Reflection\ParametersAcceptor;
+
+final class NamedArgumentsHelper
+{
+
+	public static function reorderFuncArguments(
+		ParametersAcceptor $parametersAcceptor,
+		FuncCall $functionCall
+	): FuncCall
+	{
+		return new FuncCall(
+			$functionCall->name,
+			self::reorderArgs($parametersAcceptor, $functionCall),
+			$functionCall->getAttributes()
+		);
+	}
+
+	public static function reorderMethodArguments(
+		ParametersAcceptor $parametersAcceptor,
+		MethodCall $methodCall
+	): MethodCall
+	{
+		return new MethodCall(
+			$methodCall->var,
+			$methodCall->name,
+			self::reorderArgs($parametersAcceptor, $methodCall),
+			$methodCall->getAttributes()
+		);
+	}
+
+	public static function reorderStaticCallArguments(
+		ParametersAcceptor $parametersAcceptor,
+		StaticCall $staticCall
+	): StaticCall
+	{
+		return new StaticCall(
+			$staticCall->class,
+			$staticCall->name,
+			self::reorderArgs($parametersAcceptor, $staticCall),
+			$staticCall->getAttributes()
+		);
+	}
+
+	/**
+	 * @return array<int, Arg>
+	 */
+	private static function reorderArgs(ParametersAcceptor $parametersAcceptor, CallLike $callLike): array
+	{
+		$signatureParameters = $parametersAcceptor->getParameters();
+		$callArgs = $callLike->getArgs();
+
+		$reorderedArgs = [];
+		$argumentPositions = [];
+		foreach ($signatureParameters as $i => $parameter) {
+			$argumentPositions[$parameter->getName()] = $i;
+
+			if ($parameter->isOptional()) {
+				$reorderedArgs[$i] = new Arg(
+					$parameter->getDefaultValue(),
+				);
```

Comment:
hmm it seems this is not the right place to get the default value from, since I need an `Expr` which can be passed into `Arg`'s constructor.
also: Arg needs a definitiv value and `getDefaultValue` as used here might return null, which `Arg` does not accept

Reply:
I am also wondering why `$parameter->getDefaultValue()` returns `null` for the `flags` parameter, even though [per php.net manual](https://www.php.net/json_encode) it should default to `0`

Reply:
Maybe we're not currently reading this information. We should make sure it's read from https://github.com/phpstan/php-8-stubs and phpstorm-stubs first - both have it correct. Ideally in a separate PR.